# Logfile created on 2016-11-19 13:02:12 by logger.rb/47272
I, [2016-11-19T13:02:13.200340 #11991]  INFO -- : *** below you find the most recent exception thrown, this will be likely (but not certainly) the exception that made the application exit abnormally ***
E, [2016-11-19T13:02:13.200440 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:923:in `init_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:254:in `start'
/root/coinroots/lib/daemons/amqp_daemon.rb:17:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
I, [2016-11-19T13:02:13.286805 #11991]  INFO -- : *** below you find all exception objects found in memory, some of them may have been thrown in your application, others may just be in memory because they are standard exceptions ***
E, [2016-11-19T13:02:13.287421 #11991] ERROR -- : stream closed (IOError)

E, [2016-11-19T13:02:13.287556 #11991] ERROR -- : failed to allocate memory (NoMemoryError)

E, [2016-11-19T13:02:13.287591 #11991] ERROR -- : stack level too deep (SystemStackError)

E, [2016-11-19T13:02:13.287709 #11991] ERROR -- : exception reentered (fatal)

E, [2016-11-19T13:02:13.377706 #11991] ERROR -- : Value will be available once the pipeline executes. (Redis::FutureNotReady)

E, [2016-11-19T13:02:13.442963 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/open-uri.rb:36:in `initialize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/open-uri.rb:36:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/open-uri.rb:36:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:628:in `open_logfile'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:584:in `initialize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:318:in `new'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:318:in `initialize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:353:in `new'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:353:in `exception_log'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:165:in `block in start_load'
E, [2016-11-19T13:02:13.449202 #11991] ERROR -- : No such file or directory @ rb_sysopen - /root/coinroots/log/peatio:amqp:market_data.log (Errno::ENOENT)
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/open-uri.rb:36:in `initialize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/open-uri.rb:36:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/open-uri.rb:36:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:628:in `open_logfile'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:584:in `initialize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:318:in `new'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/logger.rb:318:in `initialize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:353:in `new'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:353:in `exception_log'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:165:in `block in start_load'
E, [2016-11-19T13:02:13.450316 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `join'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `block in join'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `each'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `join'
/root/coinroots/lib/daemons/amqp_daemon.rb:81:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.451172 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:94:in `raise'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:94:in `raise'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:796:in `maybe_shutdown_reader_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:705:in `clean_up_on_shutdown'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:698:in `clean_up_and_fail_on_connection_close!'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:533:in `handle_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:81:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451235 #11991] ERROR -- : Bunny::ShutdownSignal (Bunny::ShutdownSignal)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:94:in `raise'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:94:in `raise'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:796:in `maybe_shutdown_reader_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:705:in `clean_up_on_shutdown'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:698:in `clean_up_and_fail_on_connection_close!'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:533:in `handle_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:81:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451268 #11991] ERROR -- : Connection-level error: CONNECTION_FORCED - broker forced connection closure with reason 'shutdown' (Bunny::ConnectionForced)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `join'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `block in join'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `each'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/consumer_work_pool.rb:55:in `join'
/root/coinroots/lib/daemons/amqp_daemon.rb:81:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.451311 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451340 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451384 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451447 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451511 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451550 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451618 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451660 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.451714 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `_read_from_socket'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:41:in `gets'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:273:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:229:in `block in read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:220:in `io'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:228:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `block in call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:201:in `block (2 levels) in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:309:in `ensure_connected'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:191:in `block in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:270:in `logging'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:190:in `process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:676:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `block in synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:672:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/interface.rb:9:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/ttl.rb:8:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:29:in `_marshal'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:163:in `write_entry'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:36:in `block in write'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.0.12/lib/active_support/cache.rb:519:in `instrument'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:34:in `write'
/root/coinroots/app/models/worker/market_ticker.rb:52:in `initialize_market_data'
/root/coinroots/app/models/worker/market_ticker.rb:11:in `block in initialize'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `each'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `initialize'
/root/coinroots/app/models/amqp_config.rb:25:in `new'
/root/coinroots/app/models/amqp_config.rb:25:in `binding_worker'
/root/coinroots/lib/daemons/amqp_daemon.rb:36:in `block in <top (required)>'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `each'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.451794 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `_read_from_socket'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:41:in `gets'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:273:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:229:in `block in read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:220:in `io'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:228:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `block in call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:201:in `block (2 levels) in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:309:in `ensure_connected'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:191:in `block in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:270:in `logging'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:190:in `process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:676:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `block in synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:672:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/interface.rb:9:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/ttl.rb:8:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:29:in `_marshal'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:163:in `write_entry'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:36:in `block in write'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.0.12/lib/active_support/cache.rb:519:in `instrument'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:34:in `write'
/root/coinroots/app/models/worker/market_ticker.rb:52:in `initialize_market_data'
/root/coinroots/app/models/worker/market_ticker.rb:11:in `block in initialize'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `each'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `initialize'
/root/coinroots/app/models/amqp_config.rb:25:in `new'
/root/coinroots/app/models/amqp_config.rb:25:in `binding_worker'
/root/coinroots/lib/daemons/amqp_daemon.rb:36:in `block in <top (required)>'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `each'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.456610 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `_read_from_socket'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:41:in `gets'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:273:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:229:in `block in read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:220:in `io'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:228:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `block in call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:201:in `block (2 levels) in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:309:in `ensure_connected'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:191:in `block in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:270:in `logging'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:190:in `process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:676:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `block in synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:672:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/interface.rb:9:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/ttl.rb:8:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:29:in `_marshal'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:163:in `write_entry'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:36:in `block in write'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.0.12/lib/active_support/cache.rb:519:in `instrument'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:34:in `write'
/root/coinroots/app/models/worker/market_ticker.rb:42:in `initialize_market_data'
/root/coinroots/app/models/worker/market_ticker.rb:11:in `block in initialize'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `each'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `initialize'
/root/coinroots/app/models/amqp_config.rb:25:in `new'
/root/coinroots/app/models/amqp_config.rb:25:in `binding_worker'
/root/coinroots/lib/daemons/amqp_daemon.rb:36:in `block in <top (required)>'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `each'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.456769 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:49:in `_read_from_socket'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:41:in `gets'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:273:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:229:in `block in read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:220:in `io'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:228:in `read'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `block in call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:201:in `block (2 levels) in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:309:in `ensure_connected'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:191:in `block in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:270:in `logging'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:190:in `process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:676:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `block in synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:672:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/interface.rb:9:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/ttl.rb:8:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:29:in `_marshal'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:163:in `write_entry'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:36:in `block in write'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.0.12/lib/active_support/cache.rb:519:in `instrument'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:34:in `write'
/root/coinroots/app/models/worker/market_ticker.rb:42:in `initialize_market_data'
/root/coinroots/app/models/worker/market_ticker.rb:11:in `block in initialize'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `each'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `initialize'
/root/coinroots/app/models/amqp_config.rb:25:in `new'
/root/coinroots/app/models/amqp_config.rb:25:in `binding_worker'
/root/coinroots/lib/daemons/amqp_daemon.rb:36:in `block in <top (required)>'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `each'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.456840 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:122:in `connect_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:122:in `connect_addrinfo'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:162:in `block in connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:160:in `each'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:160:in `each_with_index'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:160:in `connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:211:in `connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:285:in `establish_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:79:in `block in connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:257:in `with_reconnect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:78:in `connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:304:in `ensure_connected'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:191:in `block in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:270:in `logging'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:190:in `process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:676:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `block in synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:672:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/interface.rb:9:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/ttl.rb:8:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:29:in `_marshal'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:163:in `write_entry'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:36:in `block in write'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.0.12/lib/active_support/cache.rb:519:in `instrument'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:34:in `write'
/root/coinroots/app/models/worker/market_ticker.rb:42:in `initialize_market_data'
/root/coinroots/app/models/worker/market_ticker.rb:11:in `block in initialize'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `each'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `initialize'
/root/coinroots/app/models/amqp_config.rb:25:in `new'
/root/coinroots/app/models/amqp_config.rb:25:in `binding_worker'
/root/coinroots/lib/daemons/amqp_daemon.rb:36:in `block in <top (required)>'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `each'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.456931 #11991] ERROR -- : Operation now in progress - connect(2) would block (IO::EINPROGRESSWaitWritable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:122:in `connect_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:122:in `connect_addrinfo'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:162:in `block in connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:160:in `each'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:160:in `each_with_index'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:160:in `connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/connection/ruby.rb:211:in `connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:285:in `establish_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:79:in `block in connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:257:in `with_reconnect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:78:in `connect'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:304:in `ensure_connected'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:191:in `block in process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:270:in `logging'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:190:in `process'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis/client.rb:96:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:676:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `block in synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:37:in `synchronize'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-3.0.7/lib/redis.rb:672:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/interface.rb:9:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/ttl.rb:8:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `block in set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:29:in `_marshal'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-store-1.1.4/lib/redis/store/marshalling.rb:5:in `set'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:163:in `write_entry'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:36:in `block in write'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/activesupport-4.0.12/lib/active_support/cache.rb:519:in `instrument'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/redis-activesupport-4.0.0/lib/active_support/cache/redis_store.rb:34:in `write'
/root/coinroots/app/models/worker/market_ticker.rb:42:in `initialize_market_data'
/root/coinroots/app/models/worker/market_ticker.rb:11:in `block in initialize'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `each'
/root/coinroots/app/models/worker/market_ticker.rb:10:in `initialize'
/root/coinroots/app/models/amqp_config.rb:25:in `new'
/root/coinroots/app/models/amqp_config.rb:25:in `binding_worker'
/root/coinroots/lib/daemons/amqp_daemon.rb:36:in `block in <top (required)>'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `each'
/root/coinroots/lib/daemons/amqp_daemon.rb:35:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.457547 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457604 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457657 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457701 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457748 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457779 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457811 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457840 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457894 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457924 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.457984 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.458024 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:63:in `run_once'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:33:in `block in run_loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/reader_loop.rb:30:in `run_loop'
E, [2016-11-19T13:02:13.458087 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:990:in `open_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:255:in `start'
/root/coinroots/lib/daemons/amqp_daemon.rb:17:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.458131 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:990:in `open_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:255:in `start'
/root/coinroots/lib/daemons/amqp_daemon.rb:17:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.458182 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:940:in `open_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:255:in `start'
/root/coinroots/lib/daemons/amqp_daemon.rb:17:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.458236 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:940:in `open_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:255:in `start'
/root/coinroots/lib/daemons/amqp_daemon.rb:17:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.458334 #11991] ERROR -- : no such frame (ArgumentError)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `frame_binding'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:25:in `block (2 levels) in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:23:in `block in callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `open'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/binding_of_caller-0.7.2/lib/binding_of_caller/mri2.rb:21:in `callers'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/better_errors-1.1.0/lib/better_errors/core_ext/exception.rb:9:in `block in <class:Exception>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:923:in `init_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:254:in `start'
/root/coinroots/lib/daemons/amqp_daemon.rb:17:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
E, [2016-11-19T13:02:13.458387 #11991] ERROR -- : Resource temporarily unavailable - read would block (IO::EAGAINWaitReadable)
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `read_nonblock'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:41:in `block in read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `loop'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/cruby/socket.rb:40:in `read_fully'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/transport.rb:199:in `read_next_frame'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:923:in `init_connection'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/bunny-1.2.1/lib/bunny/session.rb:254:in `start'
/root/coinroots/lib/daemons/amqp_daemon.rb:17:in `<top (required)>'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:203:in `start_load'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/application.rb:298:in `start'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/controller.rb:70:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:147:in `block in run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `call'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons/cmdline.rb:109:in `catch_exceptions'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-1.1.9/lib/daemons.rb:146:in `run'
/root/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/daemons-rails-1.2.1/lib/daemons/rails.rb:16:in `run'
lib/daemons/market_data_ctl:11:in `<main>'
